README.md

```diff
@@ -254,15 +268,95 @@ s := repo.Subscribers().All(func(e eventsourcing.Event) {
 s.Close()
 ```
 
-## Custom made event store
+## Snapshot
 
-If you want to store your events in another database beside the already implemented event stores: `sql`, `bbolt`, `esdb` or `memory`, you can implement a custom made event store. It has to implement the following  interface to support the eventsourcing.Repository.
+If an aggregate has alot of events it can take some time fetching it's event and building the aggregate. This can be optimised with the help of a snapshot. The snapshot is the state of the aggregate on a specific version. Instead of iterating all aggregate events only the events after the version is iterated and used to build the aggregate. The use of snapshots is optional and is exposed via the snapshot repository.
+
+### Snapshot Repository
+
+The snapshot repository is used to fetch and save aggregates based snapshots and events (if there is events after the snapshot version). 
+
+The snapshot repository is a layer on top of the event repository.
 
 ```go
-type EventStore interface {
-    Save(events []core.Event) error
-    Get(id string, aggregateType string, afterVersion core.Version) (core.Iterator, error)
+NewSnapshotRepository(snapshotStore core.SnapshotStore, eventRepo *EventRepository) *SnapshotRepository
+```
+
+```go
+// fetch the aggregate baesd on it's snapshot and the events after the version of the snapshot
+GetWithContext(ctx context.Context, id string, a aggregate) error
+
+// only fetch the aggregate snapshot and not any events.
+GetSnapshot(ctx context.Context, id string, a aggregate) error
+
+// Store the aggregate events and after the snapshot
+Save(a aggregate) error
+
+// Store only the aggregate snapshot. Will return an error if there are events that are not stored on the aggregate
+SaveSnapshot(a aggregate) error
+
+// expose the underlaying event repository.
+EventRepository() *EventRepository
```

Comment:
Is `EventRepository` an interface? If so, does it make sense to return a pointed to an interface implementation (which already may be a pointer type)?

Reply:
NVM, it's a struct